lib/service/src/main/java/io/atlasmap/service/MappingService.java

```diff
@@ -91,6 +96,105 @@ public MappingService(AtlasService parent) {
         this.previewContext = this.atlasService.getContextFactory().createPreviewContext();
     }
 
+    /**
+     * Return the field from the specified list of mapped fields whose index property matches the
+     * specified field index.
+     *
+     * @param mappedFields
+     * @param fieldIndex
+     *
+     * @return field
+     */
+    private Field getFieldByIndex(List<Field> mappedFields, int fieldIndex) {
+        for (Field field : mappedFields) {
+            if (field.getIndex().intValue() == fieldIndex) {
+                return field;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return the ordinal position of the mapped field with the specified index property.
+     *
+     * @param mappedFields
+     * @param fieldIndex
+     *
+     * @return ordinal position
+     */
+    private int getOrdinalPosition(List<Field> mappedFields, int fieldIndex) {
+        int ordinalPosition = 0;
+        for (Field field : mappedFields) {
+            if (field.getIndex().intValue() == fieldIndex) {
+                break;
+            }
+            ordinalPosition++;
+        }
+        return ordinalPosition;
+    }
+
+    /**
+     * Change the value of the specified current index to the specified target index for the specified
+     * mapped field (source/target).
+     *
+     * @param mappingDefinitionId
+     * @param mappingIndex
+     * @param sourceMapping
+     * @param currentFieldIndex
+     * @param targetFieldIndex
+     *
+     * @return empty response
+     */
+    @POST
+    @Path("/{mappingIndex}/inptu/{sourceMapping}/current/{currentFieldIndex}/target/{targetFieldIndex}")
```

Comment:
I'd suggest `/{mappingIndex}/field/{dataSourceType}/{currentFieldIndex}/index` as an URL, then send the `targetFieldIndex` in the request body. We already have `DataSourceType` enum to indicate if it's `SOURCE` or `TARGET`.